### ISSUE TITLE
Rework on Audits component

### ISSUE THREAD

**Issue by hdurix** [jhipster/generator-jhipster#6974]
##### **Overview of the issue**

The audits view shows which user connected to the application and when. You can filter by date and you can sort by all fields (timestamp, principal, type, data.message).
However, only the events for the current page are sorted because the table is sorted in the Angular component: see https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/app/admin/audits/_audits.component.ts#L55.

##### **Motivation for or Use Case**

This behavior is very disturbing for the app administration. The main use case when checking the audit event logs is for me to sort them by dates. When you sort the table, you can't imagine the data is sorted only for the current page, so for example you may think the last events were done many days ago (and not see the real last events).

##### **Reproduce the error**

Just create an app, login many times (> 20 times, or change the pagination constants), go to `http://localhost:8080/#/audits` and try to sort the table. Notice you have only the data for the current page.

##### **Suggest a Fix**

The sort should be done by the server (the request is already pageable so we just have to specify the sort param in the web service).
But I reckon that all this page should be modified to be similar to other CRUD list views (entity with pagination / user management) because all the mechanism is already done in those pages and the generated app would be more homogeneous (so less code, bugs and maintenance burden).

Concerning the different front-end frameworks:
- Angular: should be done first
- AngularJS: could stay as it is
- React: filters/sort part seems not implemented yet, and it would be simpler to do it with this solution

The only regression could be for the column "Extra data" as it would be not easy to sort the events by "data.message" (`data` is a map stored in an other table) but maybe it's not a big issue if the table is not sortable by this column.

What do you think?

I can do the Pull Request for that.
It could also be a "good first issue" if someone is interested. 

##### **JHipster Version(s)**

All.

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
@hdurix it would be good to have so if you can do PR its nice, I dont think anyone else will have bandwidth from core team and I dont want it left open expecting someone to do it

**Comment by hdurix** [comment]
I'm currently working on this, quite a lot of changes, so maybe it wasn't a good idea for a first pr anyway.
